[team-11 이노] FE, iOS에 따른 GitHub OAuth 구현, 프로젝트 API 구현

eNoLJ opened the pull request:
새롭게 구현한 부분
- FE, iOS 각 어플리케이션에서 login 요청시 typeCode를 파라미터로 받아 구분하여 처리 하였습니다.
- JDBC Template을 사용하여 DAO를 만들어 사용했습니다.

이번에 처음 JDBC Template을 사용하면서 궁금한 점이 하나 생겼습니다.

DAO에서 쿼리에 해당하는 엔티티 클래스로 매핑을 진행했습니다. 여기서 궁금한 점이 생겼는데 테이블 join 시 컬럼이 늘어나게 되고 매핑 가능한 엔티티 클래스가 없어집니다. 이 때 엔티티 클래스가 아닌 별도의 클래스를 만들어 매핑을 진행하는 것이 좋은지 아니면 엔티티 클래스로 매핑하여 가져 온 뒤 코드 상에서 변환을 해주는 것이 좋은지 의견 주시면 감사하겠습니다.

전자의 경우 필요한 정보들을 얻기 위해 여러 DAO를 사용해야하는 단점이 있고, 후자의 경우 한번의 쿼리로 필요한 정보들을 얻을 수 있다는 장점이 있는 반면 엔티티 클래스가 아닌 다른 클래스가 db와 접촉을 한다는게 찝찝합니다. 그리고 그렇게 되면 자연스레 비지니스 로직 또한 매핑을 위해 새로 만든 별도의 클래스에 생기게 되는데 비지니스 로직은 service와 entity 에서 처리를 해줘야하는 걸로 알고 있어서 더 헷갈리네요.

지난 프로젝트에서는 API 만들어 지는게 늦어져 PR을 못 보냈기 때문에 엄청 오랜만에 PR을 보내게 되었네요.
OAuth든 JDBC Template이든 새롭게 적용한 것들이 많아 부족한 부분이 많으리라 생각됩니다. 사소한거 하나하나 다 지적해주시면 최대한 다 반영하도록 하겠습니다.
이번 리뷰도 잘 부탁드립니다!!

eNoLJ committed:
Feat: [#10] .gitignore 파일 추가

eNoLJ committed:
Merge pull request #12 from HongzCloud/BE/feature/setting

Spring boot 프로젝트 생성

eNoLJ committed:
Feat: [#14] HouseController 클래스에 searchHouses 메소드 생성

숙소 검색 요청에 대한 목업 API를 제공하기 위해 searchHouses 메소드 생성, 관련 dto 클래스 생성

eNoLJ committed:
Feat: [#16] HouseController 클래스에 searchCharges 메소드 생성

범위 내 숙소 가격 요청에 대한 목업 API를 제공하기 위해 searchCharges 메소드 생성, 관련 DTO 클래스 생성

eNoLJ committed:
Feat: [#18] HouseController 클래스에 getReservationPage 메소드 생성

숙소 예약 페이지 요청에 대한 목업 API를 제공하기 위해 getReservationPage 메소드 생성, 관련 DTO 클래스 생성

eNoLJ committed:
Feat: [#20] HouseController 클래스에 makeReservation 메소드 생성

숙소 예약 요청에 대한 목업 API를 제공하기 위해 makeReservation 메소드 생성, 관련 DTO 클래스 생성

eNoLJ committed:
Feat: [#21] HouseController 클래스에 getWishes 메소드 생성

위시 리스트 요청에 대한 목업 API를 제공하기 위해 getWishes 메소드 생성, 관련 DTO 클래스 생성

eNoLJ committed:
Feat: [#22] HouseController 클래스에 changeWish 메소드 생성

위시 요청에 대한 목업 API를 제공하기 위해 changeWish 메소드 생성

eNoLJ committed:
Feat: [#23] HouseController 클래스에 getReservationList 메소드 생성

숙소 예약 리스트 요청에 대한 목업 API를 제공하기 위해 getReservationList 메소드 생성, 관련 DTO 클래스 생성

eNoLJ committed:
Feat: [#24] HouseController 클래스에 getReservationDetail 메소드 생성

숙소 예약 디테일 정보 요청에 대한 목업 API를 제공하기 위해 getReservationDetail 메소드 생성, 관련 DTO 클래스 생성

eNoLJ committed:
Feat: [#25] HouseController 클래스에 cancelReservation 메소드 생성

숙소 예약 취소 요청에 대한 목업 API를 제공하기 위해 cancelReservation 메소드 생성

eNoLJ committed:
Merge pull request #26 from HongzCloud/BE/feature/mockupAPI

airbnb mockup API 생성

eNoLJ committed:
Chore: ReservationInfoResponseDTO 패키지 이동

eNoLJ committed:
Chore: .gitignore 파일에 build.sh 스크립트 추가

eNoLJ committed:
Feat: [#33] RequestLoggingFilterConfig 클래스 생성

API 요청 시 메소드와 바디가 기록에 남도록 filter 설정

eNoLJ committed:
Feat: [#33] WebConfig 클래스 생성

리액트에서 API를 사용할 수 있도록 cors 설정을 위해 WebConfig 클래스 생성

eNoLJ committed:
Chore: 불필요한 테스트 제거

eNoLJ committed:
Merge pull request #37 from HongzCloud/BE/feature/config

Config 설정

eNoLJ committed:
Refactor: DB password 변경으로 설정 변경

eNoLJ committed:
Refactor: HouseService 클래스에 가독성을 위해 makeCharges 메소드를 만들어 searchChargesByCondition 메소드 수정

eNoLJ committed:
Refactor: [#38] HouseController 클래스에 searchHouses, searchCharges 메소드의 매개변수 수정

GET 요청에 body를 담을 수 없기 때문에 Parameter로 받도록 수정

eNoLJ committed:
Merge pull request #40 from HongzCloud/BE/refactor/mockupAPI

mockup API 리팩토링

eNoLJ committed:
Feat: [#44] 유저의 정보를 저장하기 위한 스키마 생성

eNoLJ committed:
Feat: [#44] 유저의 정보를 저장하기 위해 User 클래스와 UserRepository 인터페이스 생성

eNoLJ committed:
Feat: [#44] UserController 클래스 생성

로그인, 로그아웃 기능을 제공하기 위해 UserController 클래스 생성, 관련 비즈니스 로직을 가진 UserService 클래스 생성, 관련 DTO 클래스 생성

eNoLJ committed:
Feat: [#44] GitHubOAuth 클래스 생성

getTokenAPI, getUserInfoAPI, getEmailAPI 메소드를 가지는 OAuth 인터페이스를 상속 받는 GitHubOAuth 클래스 생성, 관련 비즈니스 로직과 이넘 클래스 생성, Jwt 토큰을 사용한 인증방식 구현

eNoLJ committed:
Feat: [#44] 로그인, 로그아웃 기능 구현에 있어 필요한 예외 생성

eNoLJ committed:
Feat: [#44] Jwt를 사용하기 위해 의존성 추가

eNoLJ committed:
Feat: [#44] .gitignore 파일에 oauth.properties 파일 추가

eNoLJ committed:
Merge pull request #45 from HongzCloud/BE/feature/OAuth

로그인/로그아웃 API 생성

eNoLJ committed:
Feat: [#47] airbnb 프로젝트 schema 생성

eNoLJ committed:
Merge pull request #48 from HongzCloud/BE/feature/schema

airbnb 프로젝트 schema 생성

eNoLJ committed:
Feat: [#49] 프로젝트에 사용할 기본 데이터 삽입

eNoLJ committed:
Merge pull request #50 from HongzCloud/BE/feature/data

Feat: 프로젝트에 사용할 기본 데이터 삽입

eNoLJ committed:
Refactor: [#51] mockup API 수정

숙소 예약 페이지 요청 API와 숙소 예약 요청 API 수정

eNoLJ committed:
Merge pull request #52 from HongzCloud/BE/refactor/mockupAPI

mockup API 수정

eNoLJ committed:
Refactor: [#58] UserRepository 클래스를 JDBC Template를 사용한 UserDAO 클래스로 리팩토링

eNoLJ committed:
Feat: [#59] House, Image 엔티티 클래스, 관련 DAO 생성

숙소 검색 요청 API를 구현하기 위해 House, Image 엔티티 클래스와 HouseDAO, ImageDAO 생성

eNoLJ committed:
Feat: [#59] HouseService 클래스의 searchHousesByCondition 메소드 구현

정적 데이터만 전달하던 searchHousesByCondition 메소드를 DAO를 활용하여 실질적인 데이터를 전달하도록 구현

eNoLJ committed:
Feat: [#60] HouseService 클래스의 searchChargesByCondition 메소드 구현

정적 데이터만 전달하던 searchChargesByCondition 메소드를 DAO를 활용하여 실질적인 데이터를 전달하도록 구현

eNoLJ committed:
Feat: [#61] HouseService 클래스의 getReservationInfo 메소드 구현

정적 데이터만 전달하던 getReservationInfo 메소드를 DAO를 활용하여 실질적인 데이터를 전달하도록 구현

eNoLJ committed:
Feat: [#62] HouseService 클래스의 makeReservation 메소드 구현

정적 데이터만 전달하던 makeReservation 메소드를 DAO를 활용하여 실질적인 데이터를 전달하도록 구현

eNoLJ committed:
Refactor: UserService 클래스에 있던 getTokenFromAuthorization 메소드를 JwtUtil 클래스로 이동

eNoLJ committed:
Feat: 토큰이 null일 때 에러를 내보내도록 수정

eNoLJ committed:
Feat: [#64] HouseService 클래스의 getWishList 메소드 구현

정적 데이터만 전달하던 getWishList 메소드를 DAO를 활용하여 실질적인 데이터를 전달하도록 구현

eNoLJ committed:
Feat: [#65] HouseService 클래스의 changeWish 메소드 구현

정적 데이터만 전달하던 changeWish 메소드를 DAO를 활용하여 실질적인 데이터를 전달하도록 구현

eNoLJ committed:
Feat: [#66] HouseService 클래스의 getReservationList 메소드 구현

정적 데이터만 전달하던 getReservationList 메소드를 DAO를 활용하여 실질적인 데이터를 전달하도록 구현

eNoLJ committed:
Refactor: Authorization을 통한 인증 방법 수정

eNoLJ committed:
Feat: [#67] HouseService 클래스의 cancelReservation 메소드 구현

정적 데이터만 전달하던 cancelReservation 메소드를 DAO를 활용하여 실질적인 데이터를 전달하도록 구현

eNoLJ committed:
Feat: [#68] HouseService 클래스의 getReservationDetail 메소드 구현

정적 데이터만 전달하던 getReservationDetail 메소드를 DAO를 활용하여 실질적인 데이터를 전달하도록 구현

eNoLJ committed:
Merge pull request #69 from HongzCloud/BE/feature/API

airbnb API 구현

eNoLJ committed:
Chore: 불필요한 코드 삭제

eNoLJ committed:
Refactor: JsonProperty 어노테이션을 활용하여 리팩토링

eNoLJ committed:
Feat: UserDAO 클래스의 save 메소드 수정

save 메소드의 쿼리문을 이미 레코드 존재 시 token 값만 업데이트 하도록 수정, 불필요한 메소드 삭제, schema.sql에 userId를 유니크 키로 설정

eNoLJ committed:
Fix: 프론트의 요구사항에 맞게 API 수정

검색 요청 API의 response에 옵션 분리, local에 대한 정보 추가, 예약 페이지 요청 API에 후기 추가

eNoLJ committed:
Feat: [#77] 숙소 상세 페이지 요청 API 생성

ksundong commented on BE/src/main/java/com/enolj/airbnb/authorization/GitHubOAuth.java:
```diff
@@ -0,0 +1,77 @@
+
+import java.util.List;
+
+@PropertySource("classpath:/oauth.properties")
```
왜 프로퍼티를 직접 불러와야했나요?

eNoLJ replied:
프로퍼티를 직접 불러오지 않고 처리할 수 있는 방법이 있나요?? application.properties 파일이 아니라면 지정해 줄 필요성이 있어서 해줬습니다.

ksundong replied:
왜 별도의 프로퍼티로 관리해줘야 하나요?

eNoLJ replied:
별도의 프로퍼티로 관리 한 이유는 `application.properties` 파일은 깃헙에 올라와 있고, `oauth.properties` 파일은 `.gitignore`에 등록되어 있기 때문입니다. 저도 이 부분 때문에 깃헙 액션을 할 때 어떻게 할까 고민해봤는데 암호를 암호화하는 방법이 있더라구요. 그래서 이번 리뷰에는 적용하지 못하더라고 다음 프로젝트에는 암호화하여 `application.properties` 파일 하나로 관리하도록 하겠습니다.

ksundong replied:
그랬군요!
제가 미처 확인하지 못했었네요.
그럼 좋은 방법인 것 같습니다! 고생하셨어요.

ksundong commented on BE/src/main/java/com/enolj/airbnb/authorization/GitHubOAuth.java:
```diff
@@ -0,0 +1,77 @@
+
+@PropertySource("classpath:/oauth.properties")
+@Component
+public class GitHubOAuth implements OAuth{
```
```suggestion
public class GitHubOAuth implements OAuth {
```

ksundong replied:
OAuth라는 인터페이스의 구현체군요. 좋은 방향의 접근일 것 같습니다.

ksundong commented on BE/src/main/java/com/enolj/airbnb/authorization/GitHubOAuth.java:
```diff
@@ -0,0 +1,77 @@
+    public static final String CLIENT_ID = "client_id";
+    public static final String CLIENT_SECRET = "client_secret";
+    public static final String CODE = "code";
+    public static final String TOKEN = "token";
```
내부에서만 쓰이는 상수라면 private을, 아니라면 public을 사용해주세요.

ksundong commented on BE/src/main/java/com/enolj/airbnb/authorization/GitHubOAuth.java:
```diff
@@ -0,0 +1,77 @@
+    }
+
+    @Override
+    public TokenDTO getTokenAPI(String code, GitHubType gitHubType) {
```
getTokenAPI라는 이름이라면
TokenAPI를 가져온다는 의미가 될 것 같아요.

ksundong replied:
또한, GitHubType이라는 구현체에 종속적인 정보가 인터페이스를 통해 드러나고 있는 것 같아요.

eNoLJ replied:
`getTokenAPI` -> `tokenReceiveAPI`로 이름을 변경 했습니다.
GitHubType을 받지 않고 int형인 typeCode를 받도록 했습니다.
typeCode의 사용에 대한 근본적인 문제점은 추후 해결하도록 하겠습니다.

ksundong replied:
API라는 것을 왜 써주는 건가요?

ksundong replied:
getToken으로도 충분합니다.
API라는 불필요한 postfix를 붙여주는 이유는 무엇인가요?

eNoLJ replied:
외부 API를 호출하는 메소드라는 걸 알려주고 싶어 그렇게 이름을 지었는데 디온의 질문으로 생각해보니 호출하는 곳에서 메소드의 내부가 어떤식으로 돌아가는지는 알 필요가 없겠네요. 토큰만 잘 넘겨주면 되지.
감사합니다. 수정하도록 하겠습니다!!

ksundong replied:
네 정확합니다. 객체지향적인 관점에서는 안에서 뭘 하든 내가 요구한 것만 받을 수 있으면 됩니다.

ksundong commented on BE/src/main/java/com/enolj/airbnb/authorization/GitHubOAuth.java:
```diff
@@ -0,0 +1,77 @@
+    @Override
+    public TokenDTO getTokenAPI(String code, GitHubType gitHubType) {
+        String id = environment.getProperty(gitHubType.getClientId());
+        String secret = environment.getProperty(gitHubType.getClientSecret());
```
매번 가져와야 하는 정보일까요?

eNoLJ replied:
생성자에서 front의 id, secret과 iOS의 id, secret을 필드에 저장하도록 했습니다.
그러면서 코드가 좀 길어졌는데 보시고 조언 부탁드려요!!

ksundong replied:
한 번 확인해보겠습니다!

ksundong replied:
어디있을까요??

ksundong commented on BE/src/main/java/com/enolj/airbnb/authorization/GitHubOAuth.java:
```diff
@@ -0,0 +1,77 @@
+    public static final String TOKEN = "token";
+
+    private final RestTemplate restTemplate;
+    private final Environment environment;
```
아마 오브젝트 스타일 가이드 초반에 관련된 내용이 나올거에요.

> 서비스를 가져오는 서비스를 이용하지 말고, 사용할 서비스 그 자체를 가져와라.

여기서 말하는 서비스는 우리가 흔히 말하는 서비스는 아닙니다.
요컨데, Environment라는 무언가 값을 가져오고 조작이 가능한 것을 필드로 두지 말고, 필요한 값만 필드로 지정해두고, 생성자에서 이를 이용하도록 하는 것이 더 현명한 방법이라고 볼 수 있겠습니다.

ksundong commented on BE/src/main/java/com/enolj/airbnb/authorization/GitHubOAuth.java:
```diff
@@ -0,0 +1,77 @@
+        UriComponentsBuilder builder = UriComponentsBuilder.fromHttpUrl(GitHubUrl.ACCESS_TOKEN.getUrl())
+                .queryParam(CLIENT_ID, id)
+                .queryParam(CLIENT_SECRET, secret)
+                .queryParam(CODE, code);
```
별도의 Bean으로 만들어주는 것도 좋아보이네요.

eNoLJ replied:
별도의 Bean으로 만들어준다는 말이 해당 코드를 @Component 클래스를 만들고 메소드로 만들어 GitHubOAuth 클래스에서 주입받아 사용하라는 말씀이신가요??

ksundong replied:
Bean을 등록하는 방법은 여러가지가 있겠죠.
직접 고민해보시길 바랍니다.

ksundong replied:
제가 찍어드린 부분은 매 번 재생성될 필요가 없는 코듭니다.

ksundong replied:
`@Bean`, `@Component`는 어떤 차이가 있을까요?
어떤 상황에 사용해야할까요?

ksundong commented on BE/src/main/java/com/enolj/airbnb/authorization/GitHubOAuth.java:
```diff
@@ -0,0 +1,77 @@
+        HttpHeaders httpHeaders = new HttpHeaders();
+        httpHeaders.set(HttpHeaders.AUTHORIZATION, TOKEN + " " + token);
+        HttpEntity<?> httpEntity = new HttpEntity<>(httpHeaders);
+        List<EmailDTO> emailDTOList = restTemplate.exchange(builder.toUriString(), HttpMethod.GET, httpEntity, new ParameterizedTypeReference<List<EmailDTO>>() {}).getBody();
```
오 이런 방법이 있었군요.
저는 body를 JsonNode로 바꿔서 처리해줬던 것 같은데, 이게 더 깔끔한 것 같습니다.

ksundong commented on BE/src/main/java/com/enolj/airbnb/authorization/OAuth.java:
```diff
@@ -0,0 +1,15 @@
+import com.enolj.airbnb.web.dto.UserInfoDTO;
+import com.enolj.airbnb.web.utils.GitHubType;
+
+public interface OAuth {
```
👍

ksundong commented on BE/src/main/java/com/enolj/airbnb/config/RequestLoggingFilterConfig.java:
```diff
@@ -0,0 +1,20 @@
+public class RequestLoggingFilterConfig {
+
+    @Bean
+    public CommonsRequestLoggingFilter logFilter() {
```
오 이런 로깅하려는시도 정말 좋은 것 같아요.

ksundong replied:
다만 WebConfiguration이 있다면 그곳에 두는게 좋을 것 같아요.

ksundong commented on BE/src/main/java/com/enolj/airbnb/domain/house/House.java:
```diff
@@ -0,0 +1,181 @@
+
+public class House {
+
+    @Id
+    private Long id;
+    private String name;
+    private String description;
+    private int charge;
+    private String option;
+    private String host;
+    private double grade;
+    private int review;
+    private double latitude;
+    private double longitude;
+    private int discountRatio;
+    private int cleaningRatio;
+    private int serviceRatio;
```
필드가 너무 많은건 아닐까요?

ksundong commented on BE/src/main/java/com/enolj/airbnb/domain/house/House.java:
```diff
@@ -0,0 +1,181 @@
+                && (longitude - 0.006 <= this.longitude && this.longitude <= longitude + 0.006);
+    }
+
+    public String makeLocation() {
+        return "서초구, 서울, 한국";
+    }
+
+    public String makeLocal() {
+        return "서초구의 아파트 전체";
+    }
```
이 메서드의 용도는 짐작하기 어렵네요.

eNoLJ replied:
이 부분은 클라이언트에서 필요한 정보인데 따로 db에 담아두질 않고 하드 코딩으로 전달해주고 있습니다. 현재 프로젝트 규모에서는 저 부분을 동적으로 저장할 필요성을 느끼지 못해 채택한 방법인데 코드가 많이 지저분해졌네요...ㅠㅠ

ksundong replied:
도메인에서 관리할 정보는 아닌 것 같네요.
DTO에서 관리해주는게 어떨까요?

ksundong commented on BE/src/main/java/com/enolj/airbnb/domain/house/House.java:
```diff
@@ -0,0 +1,181 @@
+    public List<String> makeOption() {
+        List<String> splitOption = Arrays.asList(option.split(","));
+        List<String> options = new ArrayList<>();
+        options.add(String.join(" • ", splitOption.subList(0, 4)));
+        options.add(String.join(" • ", splitOption.subList(4, 8)));
```
0,4,8은 어떤 의미를 갖고 있는지 상수로 뽑아내면 좋지 않을까요?

ksundong commented on BE/src/main/java/com/enolj/airbnb/domain/house/House.java:
```diff
@@ -0,0 +1,181 @@
+        return "레지던스 전체 호스트: " + host + "님";
+    }
+
+    public String getHostImage() {
+        return "https://user-images.githubusercontent.com/63284310/119629402-f149cb00-be48-11eb-8fa5-c4415b37076d.jpeg";
+    }
```
추후에 바뀔 부분인가요?

ksundong replied:
차라리 저라면, 필드로 두고 초기화 시점에 초기화해주는 방향을 선택했을 것 같아요.

eNoLJ replied:
이 부분도 위에와 마찬가지로 현재 프로젝트 규모에서 동적으로 저장을 하지 않고 있어 하드코딩하는 형식입니다. 말씀대로 필드로 두는 방향으로 바꾸겠습니다!!

ksundong commented on BE/src/main/java/com/enolj/airbnb/domain/house/House.java:
```diff
@@ -0,0 +1,181 @@
+
+    public double getLatitude() {
+        return latitude;
+    }
+
+    public void setLatitude(double latitude) {
+        this.latitude = latitude;
+    }
+
+    public double getLongitude() {
+        return longitude;
+    }
+
+    public void setLongitude(double longitude) {
+        this.longitude = longitude;
+    }
+
+    public int getDiscountRatio() {
+        return discountRatio;
+    }
+
+    public void setDiscountRatio(int discountRatio) {
+        this.discountRatio = discountRatio;
+    }
+
+    public int getCleaningRatio() {
+        return cleaningRatio;
+    }
+
+    public void setCleaningRatio(int cleaningRatio) {
+        this.cleaningRatio = cleaningRatio;
+    }
+
+    public int getServiceRatio() {
+        return serviceRatio;
+    }
+
+    public void setServiceRatio(int serviceRatio) {
+        this.serviceRatio = serviceRatio;
+    }
```
이것이 다 필요한 지는 꼭 생각해보시기 바랍니다.

ksundong commented on BE/src/main/java/com/enolj/airbnb/domain/house/HouseDAO.java:
```diff
@@ -0,0 +1,30 @@
+    }
+
+    public Optional<House> findById(Long id) {
+        String sql = "SELECT * FROM house WHERE id = ?";
```
* 의 사용은 무조건 지양해주세요.

ksundong replied:
`*` 입니다.

ksundong commented on BE/src/main/java/com/enolj/airbnb/domain/house/HouseDAO.java:
```diff
@@ -0,0 +1,30 @@
+
+    public Optional<House> findById(Long id) {
+        String sql = "SELECT * FROM house WHERE id = ?";
+        List<House> houses = jdbcTemplate.query(sql, new BeanPropertyRowMapper<>(House.class), id);
```
👍

ksundong replied:
다만, 사용자 지정 RowMapper를 쓰는 쪽이 더 좋지 않았을까 생각해보시면 좋을 것 같아요.

ksundong commented on BE/src/main/java/com/enolj/airbnb/domain/house/HouseDAO.java:
```diff
@@ -0,0 +1,30 @@
+    public Optional<House> findById(Long id) {
+        String sql = "SELECT * FROM house WHERE id = ?";
+        List<House> houses = jdbcTemplate.query(sql, new BeanPropertyRowMapper<>(House.class), id);
+        return houses.stream().findAny();
```
stream을 굳이 사용해지 않아도 된다면, stream을 사용하지 않는 방향으로 구현해보는 것은 어떤가요?

eNoLJ replied:
stream을 사용한 이유는 객체를 optional로 만들기 위함인데 stream을 사용하지 않고 optional로 만드를 방법이 있나요?? 찾아봐도 해당 방법이 최선이였어서 선택을 했는데 있다면 좀만 더 조언 부탁드립니다!!

ksundong replied:
음 Optional을 만들기위해서 stream을 사용했다 에서 드는 의문입니다.
Optional을 직접 사용하면 안되는건가요?

ksundong replied:
`Optional.ofNullable`을 쓰면 간단해지지 않을까요?

eNoLJ replied:
`Optional.ofNullable` 해당 메소드가 있는지 몰랐네요. 스프링을 하기 전에 기본적으로 java에 대해 잘 알아야하는데 많이 부족하네요... 흠... 자바의 정석도 2권 읽다 못읽고 있는데 코쿼 끝나고라도 차근차근 구멍을 매꿔가야겠습니다.

ksundong replied:
이렇게 알면 되죠 ㅎㅎ

ksundong replied:
저는 자바 기본서 한 권도 제대로 안봤습니다.

ksundong commented on BE/src/main/java/com/enolj/airbnb/domain/house/HouseDAO.java:
```diff
@@ -0,0 +1,30 @@
+    }
+
+    public List<House> findAll() {
+        String sql = "SELECT * FROM house";
```
WHERE 절 혹은 LIMIT을 꼭 사용해주는 습관을 들이는 것을 추천드려요.
나중에 수억 수십억 건 이상의 데이터를 불러오는 경우 장애가 발생할 수도 있습니다.

ksundong commented on BE/src/main/java/com/enolj/airbnb/domain/image/Image.java:
```diff
@@ -0,0 +1,44 @@
+    @Id
+    private Long id;
+    private String url;
+    private Long houseId;
```
굳이 SpringDataJDBC 스펙에 얽매이지말고 객체지향적으로 생각했으면 어떨까? 하는 생각도 있어요.

ksundong commented on BE/src/main/java/com/enolj/airbnb/domain/join/Join.java:
```diff
@@ -0,0 +1,129 @@
+    private Long userId;
+    private Long houseId;
+
+    public Join() {}
```
빈 생성자가 존재하는 이유는 뭔가요?

ksundong commented on BE/src/main/java/com/enolj/airbnb/domain/join/Join.java:
```diff
@@ -0,0 +1,129 @@
+
+import java.time.LocalDate;
+
+public class Join {
```
Join은 의미는 알겠지만 좋은 네이밍은 아닐 것 같아요.
좀 더 생각해보는게 어떨까요?

ksundong commented on BE/src/main/java/com/enolj/airbnb/domain/join/Join.java:
```diff
@@ -0,0 +1,129 @@
+        return checkIn + " = " + checkOut;
+    }
+
+    public String getCheckInTime() {
+        return checkIn.getYear() + "년 " + checkIn.getMonthValue() + "월 " + checkIn.getDayOfMonth() + "일 오후 4:00";
+    }
+
+    public String getCheckOutTime() {
+        return checkOut.getYear() + "년 " + checkOut.getMonthValue() + "월 " + checkOut.getDayOfMonth() + "일 오후 12:00";
+    }
```
이런 출력과 관련된 로직은 도메인 보다는 DTO에 위치하는게 어떨까요?

eNoLJ replied:
위에서도 계속 지적을 해주셨는데 앞으로는 출력에 관련된 로직은 DTO에서 처리하도록 하겠습니다.
수정했습니다. 감사합니다!!

ksundong commented on BE/src/main/java/com/enolj/airbnb/domain/user/User.java:
```diff
@@ -0,0 +1,95 @@
+        this.token = token;
+    }
+
+    public static User createUser(UserInfoDTO userInfoDTO, EmailDTO emailDTO, TokenDTO tokenDTO) {
+        return new User(userInfoDTO.getUserId(), emailDTO.getEmail(), userInfoDTO.getName(), tokenDTO.getAccessToken());
+    }
```
DTO들을 불필요하게 클래스에 드러내지 말고, 값을 받아주는건 어떨까요?

eNoLJ replied:
이 말씀은 `createUser(UserInfoDTO userInfoDTO, EmailDTO emailDTO, TokenDTO tokenDTO)` 이 부분에서 DTO들로 받지 말고 하나하나의 값으로 받으란 말씀이신가요?? 근데 만약 그렇게 되면 static 메소드가 굳이 필요가 없고 new로 생성자를 이용하여 만드는게 더 효율적이라 판단되는데 말씀의 의도가 이게 맞을까요??

ksundong replied:
뭐가 더 효율적인지는 잘 모르겠습니다.
다만, DTO가 불필요하게 드러나는 형태는 별로 좋지 못한 것 같네요.

ksundong replied:
정적 팩토리 메서드는 이펙티브 자바를 한 번 읽어보시면 좋을 것 같고
생성자도 나쁜 패턴이 아니기 때문에 생성자를 적극 활용하시길 추천드립니다.

ksundong commented on BE/src/main/java/com/enolj/airbnb/errorHandler/RestTemplateResponseErrorHandler.java:
```diff
@@ -0,0 +1,26 @@
+import static org.springframework.http.HttpStatus.Series.SERVER_ERROR;
+
+@Component
+public class RestTemplateResponseErrorHandler implements ResponseErrorHandler {
```
👍

ksundong commented on BE/src/main/java/com/enolj/airbnb/service/HouseService.java:
```diff
@@ -0,0 +1,134 @@
+    }
+
+    public List<SearchResponseDTO> searchHousesByCondition(SearchRequestDTO requestDTO) {
+        return houseDAO.findAll().stream()
+                .filter(house -> house.checkCharge(requestDTO.getMinCharge(), requestDTO.getMaxCharge()))
+                .filter((house -> house.checkLocation(requestDTO.getLatitude(), requestDTO.getLongitude())))
+                .map(house -> createSearchResponseDTO(house, findOneImageByHouseId(house.getId())))
+                .collect(Collectors.toList());
```
SQL 조건문을 이용해보세요.

ksundong commented on BE/src/main/java/com/enolj/airbnb/service/HouseService.java:
```diff
@@ -0,0 +1,134 @@
+@Service
+public class HouseService {
+
+    private final UserDAO userDAO;
+    private final HouseDAO houseDAO;
+    private final ImageDAO imageDAO;
+    private final JoinDAO joinDAO;
+    private final WishDAO wishDAO;
```
동작이 한 클래스에 집중되어있습니다. 적절히 클래스를 구분지어보세요.

ksundong commented on BE/src/main/java/com/enolj/airbnb/service/HouseService.java:
```diff
@@ -0,0 +1,134 @@
+                .collect(Collectors.toList());
+    }
+
+    private Image findOneImageByHouseId(Long houseId) {
+        return imageDAO.findAll(houseId).stream()
+                .findFirst()
+                .orElseThrow(EntityNotFoundException::new);
+    }
```
stream 안에서 호출되는데 DB에 대한 요청이 있는건 별로 좋지 못한 형태라고 생각해요.

ksundong replied:
꼭 개선해주세요.

eNoLJ replied:
밑에 리뷰도 마찬가지로 이 부분은 무슨 말씀인지 잘 이해가 안갑니다. DAO를 활용하여 객체를 리스트로 받은 후 스트림으로 만들어 처리를 해주는데 여기서 stream이 db와 어떤 연관선이 있는지 잘 모르겠어요. 조금만 더 설명 부탁드립니다.

ksundong replied:
fineOneImageByHouseId는 stream안에서 호출되는 메서드입니다.
근데 그 메서드 안에 DAO로 DB에 접근하는 부분이 존재하죠.

ksundong replied:
DB를 여러번 요청하는 것 보다 한 번에 가져오는 방식을 꼭 사용해보세요.

eNoLJ replied:
아!!!!!! 인지했습니다. 무슨 말씀인지 알겠네요. 오... 그렇군요!! 이 부분은 앞으로 신경써가며 코드를 짜도록 할게요.
현재 한번에 가져오는 sql문을 사용하지 않고 간단한 crud 구문만을 사용해서 이러한 문제점들이 나오는 것 같은데 이번 리뷰에서는 적용하지 못하더라고 꼭!! 해보도록 하겠습니다.

ksundong commented on BE/src/main/java/com/enolj/airbnb/service/HouseService.java:
```diff
@@ -0,0 +1,134 @@
+    public List<ReservationResponseDTO> getReservationList(String authorization) {
+        User user = getUserFromAuthorization(userDAO, authorization);
+        return joinDAO.findAllByUserId(user.getId()).stream()
+                .map(join -> createReservationResponseDTO(findHouseById(join.getUserId()), join, findOneImageByHouseId(join.getHouseId())))
```
여기도 stream에서 DB에 연결되고 있네요.

ksundong commented on BE/src/main/java/com/enolj/airbnb/service/UserService.java:
```diff
@@ -0,0 +1,80 @@
+public class UserService {
+
+    private final UserDAO userDAO;
+    private final OAuth gitHubOAuth;
```
변수명 확인해주세요~

eNoLJ replied:
이 부분은 어떤걸 말씀하시는 걸까요??
`gitHubOAuth`를 말씀하시는 것 같은데 저게 아닌 다른 변수명은 `OAuth`, `oAuth`, `oauth` 정도인데 맞을까요??

ksundong replied:
네 맞아요~

ksundong replied:
변수명에서 구체적인 정보가 드러나기 때문에 좋지 않겠죠?

ksundong replied:
만약 여러개의 Bean이 등록된 경우라면 네이밍으로 선택해줄 수도 있고, 애노테이션으로 선택해줄 수도 있습니다.

ksundong commented on BE/src/main/java/com/enolj/airbnb/service/UserService.java:
```diff
@@ -0,0 +1,80 @@
+        userDAO.save(user);
+    }
+
+    private TokenDTO tokenRequestApi(String code, GitHubType gitHubType) {
+        return gitHubOAuth.getTokenAPI(code, gitHubType);
+    }
+
+    private UserInfoDTO userInfoRequestApi(String token) {
+        return gitHubOAuth.getUserInfoAPI(token);
+    }
+
+    private EmailDTO emailRequestApi(String token) {
+       return gitHubOAuth.getEmailAPI(token);
+    }
```
이것을 메서드 추출해서 얻을 수 있는 장점이 있었나요?
저는 해도 그만 안해도 그만으로 보이긴 합니다!

ksundong commented on BE/src/main/java/com/enolj/airbnb/service/UserService.java:
```diff
@@ -0,0 +1,80 @@
+        return userDAO.findByUserId(userId).orElseThrow(EntityNotFoundException::new);
+    }
+
+    public static User getUserFromAuthorization(UserDAO userDAO, String authorization) {
+        User user = userDAO.findByUserId(JwtUtil.getUserIdFromToken(JwtUtil.getTokenFromAuthorization(authorization))).orElseThrow(EntityNotFoundException::new);
+        if (user.getToken() == null) {
+            throw new TokenException(ErrorMessage.INVALID_TOKEN);
+        }
+        return user;
+    }
```
static 메서드는 메서드 제일 위에 올려주는 것이 관례이고
가급적이면, static 메서드의 사용을 하지 않는게 좋을 것 같아요.

ksundong commented on BE/src/main/java/com/enolj/airbnb/web/HouseController.java:
```diff
@@ -0,0 +1,80 @@
+
+    @GetMapping("/houses")
+    public List<SearchResponseDTO> searchHouses(SearchRequestDTO requestDTO) {
+        logger.info("숙소 검색 요청");
```
이런 로그는 debug가 낫습니다.

ksundong commented on BE/src/main/java/com/enolj/airbnb/web/HouseController.java:
```diff
@@ -0,0 +1,80 @@
+        houseService.changeWish(authorization, houseId);
+    }
+
+    @GetMapping("/reservation")
+    public List<ReservationResponseDTO> getReservationList(@RequestHeader String authorization) {
+        logger.info("숙소 예약 리스트 요청");
+        return houseService.getReservationList(authorization);
+    }
+
+    @GetMapping("/reservation/{houseId}")
+    public ReservationDetailDTO getReservationDetail(@RequestHeader String authorization, @PathVariable Long houseId) {
+        logger.info("{}번 숙소의 디테일 예약정보 요청", houseId);
+        return houseService.getReservationDetail(authorization, houseId);
+    }
+
+    @DeleteMapping("/reservation/{houseId}")
+    public void cancelReservation(@RequestHeader String authorization, @PathVariable Long houseId) {
+        logger.info("{}번 숙소의 예약 취소 요청", houseId);
+        houseService.cancelReservation(authorization, houseId);
+    }
```
다른 컨트롤러로 분리하면 안되나요?

eNoLJ replied:
HouseController와 HouseService 분리 했습니다. 내부 로직은 위에 DAO와 DTO, 엔티티 클래스들 수정하면서 하도록 하겠습니다.

ksundong replied:
Service로 분리가 된건가요?
저는 ReservationController를 추출하는걸 기대했는데😭

eNoLJ replied:
아 제가 잘못 말했네요
HouseController를 ReservationController와 WishController,
HouseService를 ReservationService, WishService로 분리 했습니다.

ksundong commented on BE/src/main/java/com/enolj/airbnb/web/HouseController.java:
```diff
@@ -0,0 +1,80 @@
+        return houseService.searchChargesByCondition(requestDTO);
+    }
+
+    @GetMapping("/houses/detail/{houseId}")
```
URL Path 설계를 잘 생각해보시면 좋을 것 같습니다.

eNoLJ replied:
이 부분 생각을 해봐도 마땅히 떠오르지가 않습니다.
ReservationController 클래스에도 몇가지 path가 수정되었는데 확인 부탁드려요.

현재 제일 문제가 되는게
상세 정보를 가져오는 path들이 문제가 되는 것 같은데 조언 부탁드리겠습니다. detail을 붙이는 것밖에 떠오르지가 않네요ㅠㅠ

ksundong replied:
URL 설계에 대한 가이드를 찾아보세요.
스프링 사용 초반부터 관련된 설명을 계속해서 드려서 제가 가이드를 드리기보다는
직접 학습하시고 적용하시는 것을 더 추천드립니다.

ksundong replied:
detail이 houses에 속할까요?

ksundong commented on BE/src/main/java/com/enolj/airbnb/web/HouseController.java:
```diff
@@ -0,0 +1,80 @@
+        return houseService.getReservationInfo(houseId);
+    }
+
+    @PostMapping("houses/{houseId}")
+    public void makeReservation(@RequestHeader String authorization, @PathVariable Long houseId, @RequestBody ReservationRequestDTO requestDTO) {
+        logger.info("{}번 숙소 예약 요청", houseId);
+        houseService.makeReservation(authorization, houseId, requestDTO);
+    }
```
딱 Reservation으로 알기는 좀 어려울 것 같네요.

ksundong replied:
reservation path도 있는데 그걸 이용해보면 어떨까요?

ksundong commented on BE/src/main/java/com/enolj/airbnb/web/HouseController.java:
```diff
@@ -0,0 +1,80 @@
+        houseService.makeReservation(authorization, houseId, requestDTO);
+    }
+
+    @GetMapping("/wishes")
+    public List<WishResponseDTO> getWishes(@RequestHeader String authorization) {
+        logger.info("숙소 위시 리스트 요청");
+        return houseService.getWishList(authorization);
+    }
+
+    @PostMapping("/wishes/{houseId}")
+    public void changeWish(@RequestHeader String authorization, @PathVariable Long houseId) {
+        logger.info("{}번 숙소의 위시 요청", houseId);
+        houseService.changeWish(authorization, houseId);
+    }
```
여기도 다른 컨트롤러로 분리해보시고
POST 요청에 houseId가 나오는게 맞는지 생각해보세요.

eNoLJ replied:
WishController로 분리 했습니다.
해당 api 요청 시 리소스를 새로 쓰기 때문에 POST 요청으로 했습니다. 메소드 자체는 맞는 것 같고 그렇다면 houseId가 query string으로 온게 문제라는 말씀이신 것 같은데 음... 이렇게 사용하면 안되나요?? 아님 query parameter를 사용했어야 할까요??

ksundong replied:
houseId는 QueryString으로 오지 않았습니다.
request Body로 오는 형태가 적절하겠죠.

eNoLJ replied:
그 말씀은 POST 요청에서 Query로 정보를 받지말고 저런것도 전부 body로 받는게 적절하다는 말씀으신가요??
아니면 wishes 하위에 houseId가 속하지 않으니 현재 상황에서는 적절하지 못한 것이지 `POST /houses/{houseId}` 이런식의 사용은 괜찮다는 말씀이실까요??

ksundong replied:
houseId를 POST 요청 보내는건 무엇을 생성한다는 의미인지 파악하기 힘들 수도 있지만 가능합니다.
사실 POST 요청은 다방면으로 활용하는 경우가 많기도 하고요.
다만 현재 활용하신 방법은 wishes에 속하는 houseId에 무엇을 생성한다 라는 느낌을 줍니다.
입력하신건 QueryString이 아니라 Path니까요.

ksundong commented on BE/src/main/java/com/enolj/airbnb/web/dto/Charge.java:
```diff
@@ -0,0 +1,28 @@
+package com.enolj.airbnb.web.dto;
+
+public class Charge {
+
+    private int start;
+    private int end;
```
Charge인데 어떤 start end가 어떤 의미인지 와닿지 않는 것 같아요.

eNoLJ replied:
생각해봤는데 start, end 보단 min, max가 좀 더 적절하다 생각되네요. 해당 부분은 현재로서 fe, ios에도 영향이 있어 수정보단 숙지만하고 넘어가도록 하겠습니다.
이런 디테일한 부분 언급해주시는거 너무 좋아요 감사합니다!!

ksundong commented on BE/src/main/java/com/enolj/airbnb/web/dto/Fee.java:
```diff
@@ -0,0 +1,28 @@
+package com.enolj.airbnb.web.dto;
+
+public class Fee {
+
+    private final int cleaning;
+    private final int service;
```
클래스로 추출한 시도 좋은 것 같아요.

ksundong commented on BE/src/main/java/com/enolj/airbnb/web/dto/Location.java:
```diff
@@ -0,0 +1,42 @@
+
+import com.enolj.airbnb.domain.house.House;
+
+public class Location {
+
+    private double latitude;
+    private double longitude;
```
이건 DTO일까요? VO로 관리되어야 할까요?
잘 생각해보세요.

eNoLJ replied:
dto 패키지 안에 해당 클래스 만들면서 언뜻 고민을하다 그냥 넘겼었네요.
별도의 vo 패키지를 만들어 Location, Charge, CheckDate, Description, Fee, People 클래스를 옮겼습니다.
언급한 클래스들은 dto가 아닌 vo라 판단이 되는데 맞을까요??

ksundong replied:
VO는 무엇일까요? DTO일까요? 도메인일까요? 아니면 별개의 무엇일까요?
값 객체에 대해서는 검색을 해보시면 좋은 정보를 많이 얻으실 수 있을 것 같아요.
값 객체는 말 그대로 값을 나타내는 객체입니다.
값 객체의 성질에 대해선 검색해보시구요.
값 객체인지 무엇인지는 이노의 애플리케이션이 목표하는 바에 따라 다르지만 보통 이런 좌표라던가, 돈, 날짜, 숫자 등은 값 객체로 표현될 수 있습니다.

ksundong commented on BE/src/main/java/com/enolj/airbnb/web/dto/TokenDTO.java:
```diff
@@ -0,0 +1,36 @@
+
+public class TokenDTO {
+
+    @JsonProperty("access_token")
+    private String accessToken;
+    private String token_type;
```
음.. 잘 써주시다가 왜 이렇게 쓰셨는지 궁금하네요.

eNoLJ replied:
밑에 리뷰 부분과 같은 이유인데 TokenDTO와 UserInfoDTO는 깃헙에 토큰을 요청하고, 유저정보를 요청 후 정보를 받아오는 DTO입니다. 토큰을 요청하면 깃헙에서
```json
{
  "access_token": "asdfasdfasdf"
}
```
이런식으로 보내주는데 그렇다고 필드명을 access_token으로 하자니 컨벤션에 맞지 않아 @JsonProperty를 이용하여 매핑을 해줬습니다. 밑에 리뷰의 login 부분도 마찬가지로 깃헙에서 유저 id를 저 이름으로 보내주더라구요.

ksundong replied:
token_type을 잘 보시기바랍니다.

eNoLJ replied:
오........... 이게 왜 눈에 안들어왔죠;;
번거롭게 한번 더 피드백을 하게 해드려 죄송스럽고 감사합니다.
수정했습니다.

ksundong commented on BE/src/main/java/com/enolj/airbnb/web/dto/UserInfoDTO.java:
```diff
@@ -0,0 +1,34 @@
+
+public class UserInfoDTO {
+
+    @JsonProperty("login")
+    private String userId;
```
왜 login인가요?

ksundong commented on BE/src/main/java/com/enolj/airbnb/web/utils/GitHubType.java:
```diff
@@ -0,0 +1,34 @@
+        return clientSecret;
+    }
+
+    public static GitHubType getGitHubTypeByCode(int code) {
+        switch (code) {
+            case 1: return GitHubType.FRONT;
+            case 2: return GitHubType.IOS;
+            default: throw new GitHubTypeException(ErrorMessage.INVALID_GITHUBTYPE);
+        }
+    }
```
1, 2는 하드코딩이 아닌가요?
잘 생각해보세요.

ksundong replied:
code로 관리하고자 한다면 어떻게 하는게 좋을까요?

ksundong replied:
그리고, 꼭 code로 관리해야 하는지 잘 생각해보세요.

eNoLJ replied:
제일 첫 리뷰에서 HTTP Header에 관련 정보가 있다고 하셔서 생각을 해봤는데 혹시 host가 맞을까요?? typeCode를 파라미터로 받지 않고 요청이 들어오는 host를 판별하여 프론트와 ios를 구분하면 될 것 같다고 생각을 했습니다.
의도하신 부분이 맞나요?? 원래라면 나름 적용을 해보고 질문을 하는게 맞다는건 알고 있는데 이번주 병원다니느라 많이 늦어져 질문을 먼저 하게 된 점 이해 부탁드립니다ㅠㅠ

ksundong replied:
네 맞습니다.

ksundong commented on BE/src/main/java/com/enolj/airbnb/web/utils/GitHubUrl.java:
```diff
@@ -0,0 +1,18 @@
+
+public enum GitHubUrl {
+
+    ACCESS_TOKEN("https://github.com/login/oauth/access_token"),
+    USER_INFO("https://api.github.com/user"),
+    USER_EMAIL("https://api.github.com/user/emails");
```
👍

ksundong commented on BE/src/main/java/com/enolj/airbnb/web/utils/JwtUtil.java:
```diff
@@ -0,0 +1,51 @@
+
+    public static String getTokenFromAuthorization(String authorization) {
+        String[] authArray = authorization.split(" ");
+        if (authArray.length < 2 || !authArray[0].equals("Beare")) {
```
오타가 있네요.
조심해줍시다.

ksundong commented on BE/src/main/java/com/enolj/airbnb/web/utils/JwtUtil.java:
```diff
@@ -0,0 +1,51 @@
+                    .acceptExpiresAt(600)
+                    .build();
+            return verifier.verify(token);
+        } catch (JWTDecodeException | TokenExpiredException e) {
+            throw new TokenException(ErrorMessage.INVALID_TOKEN);
+        }
```
예외 포장을 이용해보는건 어떨까요?

eNoLJ replied:
`JWTVerificationException`으로 변경했습니다.

ksundong commented on BE/src/main/java/com/enolj/airbnb/web/utils/JwtUtil.java:
```diff
@@ -0,0 +1,51 @@
+
+    public static String getTokenFromAuthorization(String authorization) {
+        String[] authArray = authorization.split(" ");
+        if (authArray.length < 2 || !authArray[0].equals("Beare")) {
+            throw new TokenException(ErrorMessage.INVALID_TOKEN);
+        }
+        return authArray[1];
```
2, 0, 1 을 쓰지 않거나 하는 방법은 없을까요?

eNoLJ replied:
수정했습니다!!

ksundong reviewed (changes requested):
안녕하세요! 이노, 리뷰어 Dion입니다!
프로젝트 진행을 하느라 고생이 많으시네요.

- 먼저 FE와 iOS Type 별로 파라미터를 받아서 처리하는게 정말 최선일지 생각해봐주세요.
  HTTP Header에 이미 관련된 정보가 들어있지 않을까요?
- 조인을 한 정보는 보통 도메인 로직 처리 보다는 정보를 반환하는데 초점이 맞춰져 있습니다. 이를 통해서 뭔가 데이터를 조작하고자 한다면, 더 생각을 해보는게 좋겠죠. 왜 도메인과 실제 로직이 동떨어져있을까? 하고요.
  뭔가 어렴풋한 질문이라 답변드리기가 어렵네요. 이럴 때에는 내가 무엇을 추구하고 있는가에 집중하는게 좋아보입니다. 내가 객체지향을 추구한다면, 보다 객체지향적인 선택을 하면 되겠죠. 객체지향이 집중하고 있는 부분과 거리가 멀다면 DTO일테고, 아니라면 엔티티겠죠. 설계가 잘못되었을 가능성도 생각해봅시다.

고생하셨습니다. 리뷰 할 내용이 많아서 시간이 조금 걸렸네요.
코멘트 참고해주시고, 다음 리뷰요청을 기다리고 있겠습니다.~

eNoLJ commented:
@ksundong 안녕하세요 디온!! 항상 코드를 짜고 난 후에 찝찝하고 먼가 부족한 느낌이 드는데 디온이 리뷰를 해주시고 고민하고 적용하다보면 그 부분이 상당히 해소될 때가 많습니다. 이번에도 디온 덕분에 많이 발전하겠네요!! 벌써부터 리뷰 적용 후가 기대가 됩니다ㅎㅎ 이번에도 디테일하게 리뷰를 남겨주셔서 정말 너무너무 감사드려요!!

자기 전에 리뷰를 읽으면 생각하느라 잠을 못잘까봐 아직 하나하나 다 읽어보진 않았지만 최대한 열심히 적용해보도록 하겠습니다.
그리고 이번엔 리뷰가 꽤나 많다보니 리뷰에 코멘트를 하나하나 달지않고 간단히 해결할 수 있는건 이모지를 남기고 질문 사항이 있는 리뷰에만 코멘트를 남기도록 하겠습니다!!

eNoLJ commented:
@ksundong 이번주 내내 치아, 턱 통증 때문에 병원 쫓아 다니느라 반영이 늦어지고 있습니다ㅠㅠ 3일동안 병원을 4군데를 갔는데 정확한 원인을 못찾고 있네요... 리뷰는 더 늦어지면 안될 것 같아 오늘부터 조금씩 적용해 가고 있는데 시간이 별로 없으니 질문사항은 먼저 질문을 해놓을게요.
가능하면 오늘 답변을 해주시면 다른 부분들과 함께 최대한 빠르게 적용해보겠습니다.

eNoLJ committed:
Refactor: [#90] WebConfig 클래스에 logFilter 메소드 생성

RequestLoggingFilterConfig 클래스에 있던 logFilter 메소드를 WebConfig 클래스로 이동

eNoLJ committed:
Refactor: [#90] OAuth 인터페이스의 메소드 이름 변경

eNoLJ committed:
Refactor: [#90] GitHubOAuth 클래스 리팩토링

상수의 접근제어자 변경, Environment를 멤버변수로 갖고 있던 부분을 값만 가지도록 변경

eNoLJ committed:
Chore: data.sql의 이미지 변경

eNoLJ committed:
Merge pull request #91 from HongzCloud/BE/refactor

OAuth 리팩토링

eNoLJ committed:
Refactor: [#90] JwtUtil 클래스 리팩토링

verifyToken 메소드에서 JWTVerificationException를 사용하여 예외 포장, getTokenFromAuthorization 메소드에서 상수를 사용하지 않도록 수정

eNoLJ committed:
Merge pull request #92 from HongzCloud/BE/refactor

JwtUtil 클래스 리팩토링

eNoLJ committed:
Refactor: [#90] HouseController 분리

HouseController 클래스의 Reservation 부분과 Wish 부분을 ReservationController 클래스와 WishController 클래스로 분리

eNoLJ committed:
Refactor: [#90] HouseService 분리

HouseService 클래스의 Reservation 부분과 Wish 부분을 ReservationService 클래스와 WishService 클래스로 분리

eNoLJ committed:
Merge pull request #94 from HongzCloud/BE/refactor

HouseController, HouseService 분리

eNoLJ committed:
Refactor: [#90] OAuth 인터페이스의 메소드 이름 수정

eNoLJ committed:
Refactor: [#90] TokenDTO 클래스의 token_type 멤버변수를 tokenType으로 변경

eNoLJ committed:
Refactor: [#90] House 클래스에 불필요한 메소드 제거

makeLocation, makeHost, makeLocal, getHostImage 메소드 제거, 해당 정보들은 DTO 클래스에서 갖도록 수정

eNoLJ committed:
Refactor: [#90] DAO 클래스에서 리턴 타입이 Optional인 메소드 수정

Optional로 만들기 위해 리스트로 받아 스트림으로 만든 후 찾아 반환하던 방식에서 직접 Optional.ofNullable 메소드를 사용하여 만들어 반환하도록 수정

eNoLJ committed:
Refactor: [#90] Join 클래스에 불필요한 메소드 제거

makeDate, getCheckInTime, getCheckOutTime, makeOption 메소드 제거, 해당 정보는 각각의 DTO 클래스에서 만들어주도록 변경

eNoLJ committed:
Refactor: [#90] DTO 패키지 내 VO 분리

eNoLJ committed:
Merge pull request #98 from HongzCloud/BE/refactor

리뷰 적용

eNoLJ commented:
@ksundong 
으어어어 디온!!!
현재 적용 안된 리뷰가

```
application.properties 하나로 관리
uri 빌더 bean으로 만들어주기
user에 static 메소드 제거
URL path 설계
OAuth 판별은 host로

- dao 관련
필드 분리
사용자 지정 RowMapper 사용
stream안에 db접근 x
findAll 에는 WHERE, LIMIT 사용
```

정도인데 이렇게 주저리 말씀드리는 이유는 디온이 정말 정성스레 리뷰를 해주셨는데 다 하지 못할 수도 있을 것 같아 죄송해서 말씀드립니다.
이번에 다 못하더라고 꼭 잊지않고 다음 프로젝트에서 하나하나 적용해보도록 하겠습니다. 기회가 된다면 다음 프로젝트에서 pr 날릴 때 봐주세요!!
이번 리뷰로도 많이 배웠고 정말 너무너무 감사드립니다!!!!
